src/const_value.rs

```diff
@@ -0,0 +1,52 @@
+//! Handling wasm constant values
+
+use crate::error::Result;
+use crate::ir::Value;
+use crate::module::emit::IdsToIndices;
+use crate::module::globals::GlobalId;
+use failure::bail;
+use parity_wasm::elements::{self, Instruction};
+
+/// A constant which is produced in WebAssembly, typically used in global
+/// initializers or element/data offsets.
+#[derive(Debug)]
+pub enum Const {
```

Comment:
I'm realizing now that `Const` is a terrible name because the whole point of its existence here is that it may not be constant.

Perhaps `Initializer`?

Reply:
`ConstExprValue`?

Reply:
Hm ok, I think I'll leave as-is for now and we can change it if we get too annoyed at it. It's not so far off because the global's value will be constant on import.... right? ish?